hybrid/compartment_examples/call_sentry/main.c

```diff
@@ -0,0 +1,88 @@
+/***
+ * This example showcases two particular actions to help compartmentalization:
+ *      - reducing the bounds of the PCC (shared.s:82)
+ *      - calling a sentry from within a bounded compartment (shared.s:114)
+ *
+ * One difficulty, and capability leak, is how to return from the sentry
+ * function back to the caller with correct capabilities (i.e., PCC). In this
+ * example, we use `clr`, and manually return to it (main.c:49). However,
+ * having the `clr` available to the sentry function means it could access
+ * capabilities it should not be privy to (e.g., if the bounds of the sentry
+ * are set to only cover the specific function. In this example, this is
+ * further aggravated by the jump between assembly and `C`: the `C` function is
+ * compiled to an assembly which makes use of `lr` (which stores pointers, not
+ * capabilities), meaning we would, by default, return to a context in which we
+ * have no execution permissions (more specifically, the `PCC` is set to cover
+ * the callee, and when we return to the caller, the `PCC` is not affected due
+ * to returning to a pointer).
+ ***/
+
+#include "../../../include/common.h"
+#include "../../include/utils.h"
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#if !defined(__CHERI_CAPABILITY_WIDTH__) || defined(__CHERI_PURE_CAPABILITY__)
+#error "This example only works on CHERI hybrid mode"
+#endif
+
+// Trampoline function to ensure PCC is rebounded after we return from a sentry
```

Comment:
I think we might want to drop "unbound" and "rebound" as their intuitive meaning is easily confused with what's actually going on. It might be better to say something like "Trampoline function to restore the caller's PCC after a sentry returns" ?

Reply:
Addressed in c9a02f2c81cc1b627dd38f71d2ab44bd3113fa16.